src/main.py

```diff
@@ -63,7 +63,17 @@
 ASSISTANT_CREDENTIALS = os.path.join(VR_CACHE_DIR, 'assistant_credentials.json')
 ASSISTANT_OAUTH_SCOPE = 'https://www.googleapis.com/auth/assistant-sdk-prototype'
 
-PID_FILE = '/run/user/%d/voice-recognizer.pid' % os.getuid()
+# Grab pid file info
+USER_ID = '%d' % os.getuid()
+GROUP_ID = '%d' % os.getgid()
+PID_PATH = '/run/user/' + USER_ID
+PID_FILE = PID_PATH + '/voice-recognizer.pid'
+PID_COMMAND = 'sudo mkdir ' + PID_PATH + ' && sudo chown -R ' + USER_ID + ':' + GROUP_ID + ' ' + PID_PATH
+
+
+# Create and own pid directory if it doesn't exist
+if not os.path.exists(PID_PATH):
+    os.system(PID_COMMAND)
```

Comment:
we shoudl also ensure the pid-file directory in the systemd service. If this is run as a service, we can't run commands with sudo as the system boot up is non interactive.